rust/kernel/fs.rs

```diff
@@ -315,11 +360,26 @@ pub mod ro {
     /// The type of the inode.
     #[derive(Copy, Clone)]
     pub enum INodeType {
+        /// Named pipe (first-in, first-out) type.
+        Fifo,
+
+        /// Character device type.
+        Chr(u32, u32),
+
         /// Directory type.
         Dir,
 
+        /// Block device type.
+        Blk(u32, u32),
+
         /// Regular file type.
         Reg,
+
+        /// Symbolic link type.
+        Lnk,
+
+        /// Named unix-domain socket type.
+        Sock,
```

Comment:
Are we sure that we want to keep these short names?

Reply:
These names come from the C definitions, for example, `S_IFSOCK`, `S_IFREG`, `S_IFLNK`, etc. I just removed the `S_IF` prefix and changed the capitalisation.

The reason I did it this way was to keep a simple relationship between the two.